### ISSUE TITLE
Metrics: Instrument tree incorrectly highlights selected item after change

### ISSUE THREAD

**Issue by JamesNK** [dotnet/aspire#1262]
The metrics page has UI for selecting an instrument:
* A dropdown to select the resource to view metrics for
* Once a resource is selected, a tree of the resource's meters (first level) a instruments (second level)

Issue:
1. Select a resource. This displays the tree.
2. Select the first instrument in the tree. It is highlighted in the tree.
3. Change the resource. This updates the tree with new values. The bug is the first instrument in the tree is still selected.

The tree should reset to nothing being selected when the dropdown value changes.

Demo:
![metrics-tree-selection](https://github.com/dotnet/aspire/assets/303201/768750f3-e2b5-4e4d-ae18-c34463d1d6b2)

**Comment by JamesNK** [comment]
I think the problem is the `InitiallySelected` bindings don't force the tree to forget what is selected.

Source:
https://github.com/dotnet/aspire/blob/2b15a508b3ee5406615ffe76987dad714f0b36cb/src/Aspire.Dashboard/Components/Pages/Metrics.razor#L32-L44

A fix would be to force Blazor to re-render the tree from a blank slate. Is that possible?

@vnbaaij @tlmii Do you have any thoughts here?

**Comment by tlmii** [comment]
After digging into this a bit, I filed https://github.com/microsoft/fluentui-blazor/issues/1087 to see what the folks over there think. It appears to me like there's a disconnect in the `CurrentSelected` parameter and the underlying web component and I'm not exactly sure what to do about it.

> InitiallySelected bindings don't force the tree to forget what is selected.

This is correct, it is only checked during Initialize for the tree view item it is set on. 

I filed #1283 as a workaround to make the behavior correct. Small change and seems to make it work, though it doesn't fix the underlying binding issues from the issue I filed above.
